File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& nums, int k) {
        vector<int> mp(k);
        int sum = 0, ans = 0;
        mp[0] = 1;
        for(int num : nums)
        {
            sum = (sum + num) % k;
            sum = (sum + k) % k;
            ans += mp[sum];
            mp[sum]++;
        }
        return ans;
    }
};

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.go
func subarraysDivByK(nums []int, k int) int {
    mp, sum, ans := map[int]int {0 : 1}, 0, 0
    for _, num := range nums {
        sum = ((sum + num) % k + k) % k
        val, _ := mp[sum]
        ans += val
        mp[sum]++
    }
    return ans
}